src/app/features/section/subpages/section-details/section-details.component.ts

```diff
@@ -0,0 +1,72 @@
+import { AsyncPipe, JsonPipe, NgIf } from '@angular/common';
+import { ChangeDetectionStrategy, Component, inject, OnInit } from '@angular/core';
+import { MatButtonModule } from '@angular/material/button';
+import { MatCardModule } from '@angular/material/card';
+import { ActivatedRoute } from '@angular/router';
+import { selectLoggedUser } from '@core/store/user.selectors';
+import { Store } from '@ngrx/store';
+import { NavigationComponent, NavigationOption } from '@shared/ui';
+import { combineLatest, map } from 'rxjs';
+import { sectionDetailsActions } from '../../store/section.actions';
+import { selectSectionDetails } from '../../store/section.selectors';
+
+@Component({
+  selector: 'app-section-details',
+  standalone: true,
+  imports: [
+    NgIf,
+    AsyncPipe,
+    JsonPipe,
+    NavigationComponent,
+    MatButtonModule,
+    MatCardModule
+  ],
+  templateUrl: './section-details.component.html',
+  styleUrls: ['./section-details.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SectionDetailsComponent implements OnInit {
+  private store = inject(Store);
+  private activeRoute = inject(ActivatedRoute);
+
+  sectionDetails$ = this.store.select(selectSectionDetails);
+  loggedInUser$ = this.store.select(selectLoggedUser);
+  navigationOptions!: NavigationOption[];
+
+  userStatus$ = combineLatest([this.loggedInUser$, this.sectionDetails$]).pipe(
+    map(([loggedInUser, sectionDetails]) => {
+      if (loggedInUser?.id === sectionDetails.sectionOwner.id) {
+        return 'Jesteś właścicielem tej sekcji';
+      } else if (sectionDetails.users.some(participant => participant.id === loggedInUser?.id)) {
+        return 'Jesteś członkiem tej sekcji';
+      } else {
+        return 'Nie jesteś członkiem tej sekcji';
+      }
```

Comment:
a może żeby było trochę bardziej czytelnie, to wydzielmy do zmiennych to sprawdzanie, czy jest ownerem, czy uczesntikiem?

const isSectionOwner = loggedInUser?.id === sectionDetails.sectionOwner.id;
 const isSectionMember = sectionDetails.users.some(participant => participant.id === loggedInUser?.id);
 
 i nizej  if (isSectionOwner) {..}

Reply:
moim zdaniem tak jak jest jest dobrze

Reply:
sugestia eweliny daje nam dużo większą czytelność kodu, nawet jeśli wykonujemy obliczenia, które wcale nie muszą się wykonać, jak w przypadku gdy wejdziemy w od razu w logikę pierwszego if'a.

jednak mikrooptymalizacje ze względu na performance kosztem czytelności, gdy nie ma jeszcze tak naprawdę wielkiego wpływu te działanie na performance aplikacji powoduje że rośnie nam koszt utrzymania tego kodu. 

Proponuję wynieść zatem te poszczególnie działania do zmiennych.